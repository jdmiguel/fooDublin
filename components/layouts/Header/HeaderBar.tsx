import { Breadcrumbs } from '../../core/Breadcrumbs/Breadcrumbs';
import { CustomLink } from '../../core/CustomLink/CustomLink';
import { StyledHeaderBarWrapper, StyledHeaderBar } from './styles';
import { BreadcrumbsData } from '../../core/types';

type HeaderBarProps = {
  onClickBreadcrumb: (route: string, asRoute: string) => void;
  onClickFavorites: () => void;
  breadcrumbs: BreadcrumbsData[];
};

export const HeaderBar: React.FC<HeaderBarProps> = ({ onClickBreadcrumb, onClickFavorites, breadcrumbs }) => (
  <StyledHeaderBarWrapper data-testid="header-bar">
    <StyledHeaderBar className="grid-container">
    <Breadcrumbs breadcrumbsData={breadcrumbs || []} onClickBreadcrumb={onClickBreadcrumb} />
      <CustomLink onClick={onClickFavorites}>
        <i className="material-icons">bookmarks</i>FAVORITES
      </CustomLink>
    </StyledHeaderBar>
  </StyledHeaderBarWrapper>
);
