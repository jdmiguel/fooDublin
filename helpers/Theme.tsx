import React from 'react';
import { DefaultTheme, ThemeProvider } from 'styled-components';

export const theme: DefaultTheme = {
  palette: {
    PRIMARY: '#EC7E00',
    PRIMARY_DARK: '#D48C24',
    PRIMARY_MEDIUM: '#FBA52B',
    PRIMARY_LIGHT: '#FDF3E5',
    SECONDARY: '#660000',
    LIGHT_MAX: '#FFFFFF',
    LIGHT_MEDIUM: '#F3F3F3',
    LIGHT_MIN: '#E5EAED',
    LIGHT_SOFT: '#D8D8D8',
    DARK_MAX: '#222C3C',
    DARK_MEDIUM: '#585858',
    DARK_MIN: '#89959B',
    DARK_SOFT: '#A7A7A7',
  },
};

export const renderWithTheme = (component: any) => (
  <ThemeProvider theme={theme}>{component}</ThemeProvider>
);
