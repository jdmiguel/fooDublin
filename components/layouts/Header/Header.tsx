import React from 'react';
import Link from 'next/link';
import styled, { css } from 'styled-components';

import { Logo }, { LogoSize } from '../../core/Logo/Logo';
import { CustomLink }, { CustomLinkSize } from '../../core/CustomLink/CustomLink';

import Finder from '../../ui/Finder/Finder';

import { CDN_URL_STATIC_DIRECTORY } from '../../../helpers/utils';

type HeaderProps = {
  claimTxt: string;
  bgImgSrc?: string | undefined;
  isExtended: boolean;
};

const StyledHeader = styled.header<{ bgImg: string | undefined }>`
  width: 100%;
  display: flex;
  justify-content: center;
  border-bottom: 1px solid ${({ theme }) => theme.palette.LIGHT_SOFT};
  background-image: url(${({ bgImg }) => bgImg && bgImg});
  background-color: ${({ theme }) => theme.palette.DARK_SOFT};
  background-size: cover;
  background-position: center;
  display: flex;
`;

const StyledOverlay = styled.div<{ isExtended: boolean }>`
  width: 100%;
  height: ${({ isExtended }) => (isExtended ? '550px' : '80px')};
  background: rgba(0, 0, 0, 0.8);
`;

const headerContentCSS = css`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

const basicHeaderContentCSS = css`
  display: flex;
  justify-content: space-around;
  align-items: center;
  max-width: 1140px;
  @media only screen and (min-width: 1200px) {
    justify-content: space-between;
  }
`;

const StyledHeaderContent = styled.div<{ isExtended: boolean }>`
  width: 100%;
  height: 100%;
  ${({ isExtended }) =>
    isExtended ? headerContentCSS : basicHeaderContentCSS};
  margin: ${({ isExtended }) => (isExtended ? '15px 0 25px' : '0 auto')};
`;

const StyledHeaderClaim = styled.h2<{ isExtended: boolean }>`
  font-size: ${({ isExtended }) => (isExtended ? '1.25rem' : '1.3rem')};
  text-align: center;
  font-weight: 400;
  color: ${({ theme }) => theme.palette.DARK_SOFT};
  margin-bottom: ${({ isExtended }) => isExtended && '35px'};
  display: ${({ isExtended }) => !isExtended && 'none'};
  @media only screen and (min-width: 350px) {
    font-size: ${({ isExtended }) => isExtended && '1.45rem'};
  }
  @media only screen and (min-width: 768px) {
    display: ${({ isExtended }) => !isExtended && 'block'};
    font-size: ${({ isExtended }) => isExtended && '2rem'};
  }
  @media only screen and (min-width: 992px) {
    font-size: ${({ isExtended }) => (isExtended ? '2.5rem' : '1.6rem')};
  }
`;

const StyledCustomLink = styled(CustomLink)`
  @media only screen and (max-width: 539px) {
    color: ${({ theme }) => theme.palette.LIGHT_MAX};
    padding: 10px;
    border-radius: 50%;
    background-color: ${({ theme }) => theme.palette.PRIMARY_MEDIUM};
    i {
      margin-right: 0;
    }
  }
`;

const StyledCustomLinkText = styled.span`
  display: none;
  @media only screen and (min-width: 540px) {
    display: block;
  }
`;

const Header: React.FC<HeaderProps> = ({ bgImgSrc, claimTxt, isExtended }) => (
  <StyledHeader data-testid="header" bgImg={bgImgSrc}>
    <StyledOverlay isExtended={isExtended}>
      <StyledHeaderContent isExtended={isExtended}>
        <Link href="/">
          <a>
            <Logo
              size={isExtended ? LogoSize.BIG : LogoSize.SMALL}
              logoSrc={`${CDN_URL_STATIC_DIRECTORY}/images/logo.svg`}
            />
          </a>
        </Link>
        <StyledHeaderClaim isExtended={isExtended}>
          {claimTxt}
        </StyledHeaderClaim>
        {isExtended && <Finder />}
        {!isExtended && (
          <Link href="/favorites">
            <StyledCustomLink size={CustomLinkSize.BIG}>
              <i className="material-icons">bookmarks</i>
              <StyledCustomLinkText>FAVORITES</StyledCustomLinkText>
            </StyledCustomLink>
          </Link>
        )}
      </StyledHeaderContent>
    </StyledOverlay>
  </StyledHeader>
);

export default Header;
