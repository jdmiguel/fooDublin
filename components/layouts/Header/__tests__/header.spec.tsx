import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { render } from '@testing-library/react';

import reducer from '../../../../store/reducer';

import { Header } from '../Header';

import { HEADER_TEXT_MOCKS } from '../__mocks__/header.mocks';
import { renderWithTheme } from '../../../../helpers/Theme';

const { bgImgSrc, claimTxt } = HEADER_TEXT_MOCKS;

describe('Component: Header', () => {
  it('should render extended Header', () => {
    const mockStore = configureStore();
    const store = mockStore({
      reducer,
    });
    const { container } = render(
      renderWithTheme(
        <Provider store={store}>
          <Header
            bgImgSrc={`/images/${bgImgSrc}`}
            claimTxt={claimTxt}
            isExtended={true}
          />
        </Provider>,
      ),
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render Header', () => {
    const { container } = render(
      renderWithTheme(
        <Header
          bgImgSrc={`/images/${bgImgSrc}`}
          claimTxt={claimTxt}
          isExtended={false}
        />,
      ),
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
