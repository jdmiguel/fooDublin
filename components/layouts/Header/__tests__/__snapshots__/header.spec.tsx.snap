// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Header should render collapsed Header 1`] = `
.c4 {
  width: 190px;
}

.c5 {
  max-width: 300px;
}

.c3 {
  font-size: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 600;
  cursor: pointer;
  color: #FBA52B;
}

.c3 i {
  font-size: 0.8em;
  margin-right: 4px;
}

.c7 {
  font-size: 1.3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 600;
  cursor: pointer;
  color: #FBA52B;
}

.c7 i {
  font-size: 0.8em;
  margin-right: 6px;
}

.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-bottom: 1px solid #D8D8D8;
  background-image: url(/images/food.jpg);
  background-color: #A7A7A7;
  background-size: cover;
  background-position: center;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  z-index: 1;
}

.c1 {
  width: 100%;
  background: rgba(0,0,0,0.8);
  height: 80px;
}

.c2 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 1140px;
  margin: 0 auto;
}

.c6 {
  font-size: 1.3rem;
  text-align: center;
  font-weight: 400;
  color: #A7A7A7;
  display: none;
}

.c9 {
  display: none;
}

@media only screen and (min-width:768px) {

}

@media only screen and (min-width:992px) {
  .c4 {
    width: 200px;
  }
}

@media only screen and (min-width:540px) {
  .c3 {
    -webkit-transition: color 0.2s ease-out;
    transition: color 0.2s ease-out;
  }

  .c3:hover {
    color: #EC7E00;
  }
}

@media only screen and (min-width:540px) {
  .c7 {
    -webkit-transition: color 0.2s ease-out;
    transition: color 0.2s ease-out;
  }

  .c7:hover {
    color: #EC7E00;
  }
}

@media only screen and (min-width:992px) {
  .c1 {
    height: 80px;
  }
}

@media only screen and (min-width:1200px) {
  .c2 {
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

@media only screen and (min-width:350px) {

}

@media only screen and (min-width:768px) {
  .c6 {
    display: block;
  }
}

@media only screen and (min-width:992px) {
  .c6 {
    font-size: 1.6rem;
  }
}

@media only screen and (max-width:539px) {
  .c8 {
    color: #FFFFFF;
    padding: 10px;
    border-radius: 50%;
    background-color: #FBA52B;
  }

  .c8 i {
    margin-right: 0;
  }
}

@media only screen and (min-width:540px) {
  .c9 {
    display: block;
  }
}

<header
  class="c0"
  data-testid="header"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <a
        class="c3"
      >
        <h1
          class="c4"
          data-testid="logo"
        >
          <img
            alt="FooDublin Logo"
            class="c5"
            src="/images/logo.svg"
          />
        </h1>
      </a>
      <h2
        class="c6"
      >
        Mocked claim text
      </h2>
      <a
        class="c7 c8"
      >
        <i
          class="material-icons"
        >
          bookmarks
        </i>
        <span
          class="c9"
        >
          FAVORITES
        </span>
      </a>
    </div>
  </div>
</header>
`;

exports[`Component: Header should render extended Header 1`] = `
.c4 {
  width: 225px;
  margin: 15px 0 25px;
}

.c5 {
  max-width: 300px;
}

.c3 {
  font-size: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 600;
  cursor: pointer;
  color: #FBA52B;
}

.c3 i {
  font-size: 0.8em;
  margin-right: 4px;
}

.c10 {
  width: 100%;
  border-radius: 4px;
  padding: 0 20px;
  background: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c10 i {
  font-size: 1.2rem;
  color: #A7A7A7;
  margin-right: 10px;
}

.c12 {
  width: 100%;
  height: 55px;
  padding: 10px 0;
  line-height: 20px;
  border: none;
  outline: none;
  color: #2f3235;
  background: transparent;
  cursor: pointer;
  outline: none;
  -webkit-tap-highlight-color: transparent;
}

.c23 {
  color: #585858;
  font-size: 1.2rem;
  font-weight: 500;
}

.c8 {
  width: 100%;
  max-width: 550px;
  height: 55px;
  position: relative;
  cursor: pointer;
}

.c11 {
  background-color: #F3F3F3;
  -webkit-transition: background-color 0.2s ease-out;
  transition: background-color 0.2s ease-out;
}

.c11:hover {
  background-color: #FDF3E5;
}

.c13 {
  visibility: hidden;
  opacity: 0;
  -webkit-transform: translateY(10px);
  -ms-transform: translateY(10px);
  transform: translateY(10px);
  -webkit-transition: opacity 0.2s ease 0s,-webkit-transform 0.2s ease 0s;
  -webkit-transition: opacity 0.2s ease 0s,transform 0.2s ease 0s;
  transition: opacity 0.2s ease 0s,transform 0.2s ease 0s;
  position: absolute;
  box-sizing: border-box;
  background: #F3F3F3;
  z-index: 2;
  top: 55px;
  left: 0;
  padding: 0;
  width: 100%;
  height: auto;
  max-height: 400px;
  overflow: auto;
  box-shadow: 0 2px 2px rgba(0,0,0,0.35);
  border: 1px solid #F3F3F3;
  border-radius: 0 0 4px 4px;
  border-bottom-right-radius: 4px;
  border-bottom-left-radius: 4px;
  border-top: 0;
}

.c14 {
  width: 100%;
  -webkit-tap-highlight-color: transparent;
}

.c17 {
  width: 100%;
  max-width: 550px;
  height: 100%;
  position: initial;
}

.c19 {
  position: relative;
  border-radius: 4px;
  background-color: #F3F3F3;
}

.c20 {
  width: 100%;
  height: 55px;
  background-color: #F3F3F3;
  line-height: 55px;
  padding: 0 4px 0 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  cursor: pointer;
  outline: none;
  -webkit-tap-highlight-color: transparent;
  -webkit-transition: background-color 0.2s ease-out;
  transition: background-color 0.2s ease-out;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -moz-user-select: none;
}

.c20 i:first-of-type {
  font-size: 1rem;
  margin-left: 5px;
  margin-right: 10px;
  color: #D48C24;
}

.c20:hover {
  background-color: #FDF3E5;
}

.c20 i:last-of-type {
  position: absolute;
  z-index: 1;
  top: 14px;
  right: 7px;
  font-size: 1.7rem;
}

.c21 {
  visibility: hidden;
  opacity: 0;
  overflow: auto;
  -webkit-transition: opacity 0.2s ease 0s;
  transition: opacity 0.2s ease 0s;
  position: absolute;
  box-sizing: border-box;
  background-color: #F3F3F3;
  z-index: 2;
  top: 0;
  left: 0;
  padding: 20px 10px 10px;
  width: 100%;
  height: 100%;
  max-height: 100vh;
  border: 1px solid #F3F3F3;
  border-radius: 4px;
  border-top: 0;
  border-bottom: 0;
  outline: none;
}

.c22 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 10px;
  margin: 10px 0 15px;
}

.c24 {
  font-size: 1.35rem;
}

.c25 {
  cursor: pointer;
  line-height: 0;
  background-color: transparent;
  outline: none;
  -webkit-tap-highlight-color: transparent;
  margin-top: 4px;
}

.c25 i {
  font-size: 1.1rem;
  font-weight: 600;
}

.c25 i {
  color: #585858;
  font-size: 1.5rem;
}

.c26 {
  width: 100%;
  height: 55px;
  border-bottom: 1px solid #E5EAED;
  list-style: none;
  display: block;
  overflow: hidden;
  color: #2f3235;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding: 9px 11px;
  cursor: pointer;
  outline: none;
  -webkit-tap-highlight-color: transparent;
}

.c26:hover {
  background-color: #FDF3E5;
}

.c26:focus {
  background-color: #FDF3E5;
}

.c28 {
  width: 18px;
  margin-right: 12px;
}

.c27 {
  color: #2f3235;
  font-size: 1rem;
  font-weight: 400;
}

.c29 {
  width: 100%;
  height: 55px;
}

.c31 {
  width: 100%;
  height: 100%;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  text-transform: uppercase;
  padding: 0 15px;
  border-radius: 4px;
  cursor: pointer;
  outline: none;
  background-color: #FBA52B;
  color: #FFFFFF;
  font-weight: 600;
  -webkit-transition: background-color 0.2s ease-out;
  transition: background-color 0.2s ease-out;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c31 i {
  margin-right: 6px;
  font-size: 1.3em;
}

.c31:hover {
  background-color: #EC7E00;
}

.c31:focus {
  background-color: #EC7E00;
}

.c7 {
  width: 85%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c9 {
  margin: 0 0 25px;
}

.c15 {
  background-color: #A7A7A7;
  width: 100%;
  height: 1px;
  margin: 0 0 25px 0;
  max-width: 550px;
}

.c16 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 22px;
}

.c16:first-child {
  margin-right: 2%;
}

.c18 {
  width: 100%;
}

.c18:first-of-type {
  margin-right: 2%;
  margin-bottom: 15px;
}

.c30 {
  width: 100%;
  margin-bottom: 10px;
}

.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-bottom: 1px solid #D8D8D8;
  background-image: url(/images/food.jpg);
  background-color: #A7A7A7;
  background-size: cover;
  background-position: center;
}

.c1 {
  width: 100%;
  background: rgba(0,0,0,0.8);
  height: 550px;
}

.c2 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  font-size: 1.25rem;
  text-align: center;
  font-weight: 400;
  color: #A7A7A7;
  margin-bottom: 35px;
}

@media only screen and (min-width:768px) {
  .c4 {
    width: 270px;
  }
}

@media only screen and (min-width:992px) {
  .c4 {
    width: 300px;
    margin: 0 0 30px;
  }
}

@media only screen and (min-width:540px) {
  .c3 {
    -webkit-transition: color 0.2s ease-out;
    transition: color 0.2s ease-out;
  }

  .c3:hover {
    color: #EC7E00;
  }
}

@media only screen and (min-width:768px) {
  .c10 {
    max-width: 550px;
  }
}

@media only screen and (min-width:992px) {
  .c23 {
    font-size: 1.45rem;
  }
}

@media only screen and (min-width:768px) {
  .c17 {
    height: 55px;
    position: relative;
  }
}

@media only screen and (min-width:768px) {
  .c21 {
    max-height: 440px;
    height: auto;
    padding: 0;
    box-shadow: 0 2px 2px rgba(0,0,0,0.35);
    -webkit-transition: opacity 0.2s ease 0s,-webkit-transform 0.2s ease 0s;
    -webkit-transition: opacity 0.2s ease 0s,transform 0.2s ease 0s;
    transition: opacity 0.2s ease 0s,transform 0.2s ease 0s;
    -webkit-transform: translateY( 20px );
    -ms-transform: translateY( 20px );
    transform: translateY( 20px );
  }
}

@media only screen and (min-width:768px) {
  .c22 {
    display: none;
  }
}

@media only screen and (min-width:992px) {
  .c27 {
    font-size: 0.9rem;
  }
}

@media only screen and (min-width:470px) {
  .c7 {
    width: 70%;
  }
}

@media only screen and (min-width:768px) {
  .c7 {
    width: 60%;
  }
}

@media only screen and (min-width:992px) {
  .c7 {
    width: 58rem;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media only screen and (min-width:1250px) {
  .c7 {
    width: 71rem;
  }
}

@media only screen and (min-width:992px) {
  .c9 {
    margin: 0 2% 0 0;
    width: 39%;
  }
}

@media only screen and (min-width:1200px) {
  .c9 {
    width: 44%;
  }
}

@media only screen and (min-width:992px) {
  .c15 {
    width: 1px;
    height: 90%;
    margin: 0 2% 0 0;
  }
}

@media only screen and (min-width:768px) {
  .c16 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media only screen and (min-width:992px) {
  .c16 {
    width: 47%;
    -webkit-box-pack: space-around;
    -webkit-justify-content: space-around;
    -ms-flex-pack: space-around;
    justify-content: space-around;
    margin-bottom: 0;
  }

  .c16:first-child {
    margin-right: 0;
  }
}

@media only screen and (min-width:768px) {
  .c18 {
    max-width: 270px;
  }

  .c18:first-of-type {
    margin-bottom: 0;
  }
}

@media only screen and (min-width:992px) {
  .c18 {
    margin-right: 2%;
    width: 48%;
  }
}

@media only screen and (min-width:768px) {
  .c30 {
    width: 300px;
    margin-bottom: 0;
  }
}

@media only screen and (min-width:992px) {
  .c30 {
    width: 10%;
  }
}

@media only screen and (min-width:992px) {
  .c1 {
    height: 440px;
  }
}

@media only screen and (min-width:350px) {
  .c6 {
    font-size: 1.45rem;
  }
}

@media only screen and (min-width:768px) {
  .c6 {
    font-size: 2rem;
  }
}

@media only screen and (min-width:992px) {
  .c6 {
    font-size: 2.5rem;
  }
}

<header
  class="c0"
  data-testid="header"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <a
        class="c3"
      >
        <h1
          class="c4"
          data-testid="logo"
        >
          <img
            alt="FooDublin Logo"
            class="c5"
            src="/images/logo.svg"
          />
        </h1>
      </a>
      <h2
        class="c6"
      >
        Mocked claim text
      </h2>
      <div
        class="c7"
        data-testid="finder"
      >
        <div
          class="c8 c9"
          data-testid="autocomplete"
        >
          <div
            class="c10 c11"
            data-testid="input-wrapper"
          >
            <i
              class="material-icons"
              data-testid="input-icon"
            >
              search
            </i>
            <input
              class="c12"
              placeholder="Search for locals..."
              type="text"
              value=""
            />
          </div>
          <div
            class="c13"
            data-testid="listbox-wrapper"
          >
            <ul
              class="c14"
              role="listbox"
            />
          </div>
        </div>
        <span
          class="c15"
        />
        <div
          class="c16"
        >
          <div
            class="c17 c18"
            data-testid="dropdown"
          >
            <div
              class="c19"
            >
              <button
                class="c20"
                type="button"
              >
                <i
                  class="material-icons"
                  data-testid="label-icon"
                >
                  near_me
                </i>
                <span>
                  Select any location
                </span>
                <i
                  class="material-icons"
                >
                  arrow_drop_down
                </i>
              </button>
            </div>
            <div
              class="c21"
              data-testid="dropdown-list"
              role="listbox"
              tabindex="0"
            >
              <div
                class="c22"
              >
                <h4
                  class="c23 c24"
                >
                  Select any location
                </h4>
                <button
                  class="c25"
                  type="button"
                >
                  <i
                    class="material-icons"
                  >
                    close
                  </i>
                </button>
              </div>
              <div
                class="c26"
                role="option"
              >
                <p
                  class="c27"
                >
                  Dublin
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <p
                  class="c27"
                >
                  South City West
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <p
                  class="c27"
                >
                  Temple Bar
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <p
                  class="c27"
                >
                  Clondalkin
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <p
                  class="c27"
                >
                  Swords
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <p
                  class="c27"
                >
                  Tallaght
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <p
                  class="c27"
                >
                  Blanchardstown
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <p
                  class="c27"
                >
                  South City East
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <p
                  class="c27"
                >
                  Dundrum
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <p
                  class="c27"
                >
                  Finglas
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <p
                  class="c27"
                >
                  Ballyfermot
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <p
                  class="c27"
                >
                  Santry
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <p
                  class="c27"
                >
                  Lucan
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <p
                  class="c27"
                >
                  Artane
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <p
                  class="c27"
                >
                  Malahide
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <p
                  class="c27"
                >
                  Dun Laoghaire
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <p
                  class="c27"
                >
                  Walkinstown
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <p
                  class="c27"
                >
                  Blackrock
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <p
                  class="c27"
                >
                  Palmerstown
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <p
                  class="c27"
                >
                  Rathmines
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <p
                  class="c27"
                >
                  Balbriggan
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <p
                  class="c27"
                >
                  Ballybrack
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <p
                  class="c27"
                >
                  Smithfield
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <p
                  class="c27"
                >
                  Crumlin
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <p
                  class="c27"
                >
                  Castleknock
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <p
                  class="c27"
                >
                  Howth
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <p
                  class="c27"
                >
                  Cabra
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <p
                  class="c27"
                >
                  Whitehall
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <p
                  class="c27"
                >
                  Portmarnock
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <p
                  class="c27"
                >
                  Kimmage
                </p>
              </div>
            </div>
          </div>
          <div
            class="c17 c18"
            data-testid="dropdown"
          >
            <div
              class="c19"
            >
              <button
                class="c20"
                type="button"
              >
                <i
                  class="material-icons"
                  data-testid="label-icon"
                >
                  restaurant
                </i>
                <span>
                  Select any cuisine
                </span>
                <i
                  class="material-icons"
                >
                  arrow_drop_down
                </i>
              </button>
            </div>
            <div
              class="c21"
              data-testid="dropdown-list"
              role="listbox"
              tabindex="0"
            >
              <div
                class="c22"
              >
                <h4
                  class="c23 c24"
                >
                  Select any cuisine
                </h4>
                <button
                  class="c25"
                  type="button"
                >
                  <i
                    class="material-icons"
                  >
                    close
                  </i>
                </button>
              </div>
              <div
                class="c26"
                role="option"
              >
                <img
                  alt="Irish"
                  class="c28"
                  src="/images/stew.svg"
                />
                <p
                  class="c27"
                >
                  Irish
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <img
                  alt="Italian"
                  class="c28"
                  src="/images/pizza.svg"
                />
                <p
                  class="c27"
                >
                  Italian
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <img
                  alt="Mediterranean"
                  class="c28"
                  src="/images/bruschetta.svg"
                />
                <p
                  class="c27"
                >
                  Mediterranean
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <img
                  alt="Asian"
                  class="c28"
                  src="/images/rice.svg"
                />
                <p
                  class="c27"
                >
                  Asian
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <img
                  alt="Healthy Food"
                  class="c28"
                  src="/images/healthy_food.svg"
                />
                <p
                  class="c27"
                >
                  Healthy Food
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <img
                  alt="Indian"
                  class="c28"
                  src="/images/gulab.svg"
                />
                <p
                  class="c27"
                >
                  Indian
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <img
                  alt="Mexican"
                  class="c28"
                  src="/images/taco.svg"
                />
                <p
                  class="c27"
                >
                  Mexican
                </p>
              </div>
              <div
                class="c26"
                role="option"
              >
                <img
                  alt="Fast Food"
                  class="c28"
                  src="/images/fries.svg"
                />
                <p
                  class="c27"
                >
                  Fast Food
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="c29 c30"
        >
          <button
            class="c31"
            type="button"
          >
            Search
          </button>
        </div>
      </div>
    </div>
  </div>
</header>
`;
